iot-hub/Samples/device/AndroidSample/things/src/main/java/android/iot/sdk/azure/microsoft/com/things/MainActivity.java

```diff
@@ -0,0 +1,251 @@
+package android.iot.sdk.azure.microsoft.com.things;
+
+import android.app.Activity;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.os.Bundle;
+import android.os.Handler;
+
+import com.microsoft.azure.sdk.iot.device.DeviceClient;
+import com.microsoft.azure.sdk.iot.device.DeviceTwin.DeviceMethodData;
+import com.microsoft.azure.sdk.iot.device.IotHubClientProtocol;
+import com.microsoft.azure.sdk.iot.device.IotHubEventCallback;
+import com.microsoft.azure.sdk.iot.device.IotHubMessageResult;
+import com.microsoft.azure.sdk.iot.device.IotHubStatusCode;
+import com.microsoft.azure.sdk.iot.device.Message;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+
+/**
+ * Skeleton of an Android Things activity.
+ * <p>
+ * Android Things peripheral APIs are accessible through the class
+ * PeripheralManagerService. For example, the snippet below will open a GPIO pin and
+ * set it to HIGH:
+ * <p>
+ * <pre>{@code
+ * PeripheralManagerService service = new PeripheralManagerService();
+ * mLedGpio = service.openGpio("BCM6");
+ * mLedGpio.setDirection(Gpio.DIRECTION_OUT_INITIALLY_LOW);
+ * mLedGpio.setValue(true);
+ * }</pre>
+ * <p>
+ * For more complex peripherals, look for an existing user-space driver, or implement one if none
+ * is available.
+ *
+ * @see <a href="https://github.com/androidthings/contrib-drivers#readme">https://github.com/androidthings/contrib-drivers#readme</a>
+ */
+public class MainActivity extends Activity {
+
+    private final String connString = BuildConfig.DeviceConnectionString;;
+
+    private double temperature;
+    private double humidity;
+    private String msgStr;
+    private Message sendMessage;
+
+    private DeviceClient client;
+
+    IotHubClientProtocol protocol = IotHubClientProtocol.MQTT;
+
+    private int sendMessagesInterval = 5000;
+
+    private final Handler handler = new Handler();
+    private Thread sendThread;
+
+    private static final int METHOD_SUCCESS = 200;
+    public static final int METHOD_THROWS = 403;
+    private static final int METHOD_NOT_DEFINED = 404;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        while (!checkInternetConnection()){}
+
+        try
+        {
+            start();
+        }
+        catch (Exception e2)
+        {
+            System.out.println("Exception while opening IoTHub connection: " + e2.toString());
+
```

Comment:
Do you have to do toString for exception? It should not be needed.

Reply:
printing the whole stacktrace would be better here